### DISCUSSION TITLE
Windows App SDK - Unpackaged deploy docs \ example

### DISCUSSION THREAD

**Discussion by capegreg** [microsoft/WindowsAppSDK#1351] [Q&A] [answered]
I am looking for documentation and example to deploy a UWP app as unpackaged using Version 1.0 Experimental (1.0.0-experimental1). The example project [https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/tutorial-unpackaged-deployment?tabs=cpp](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/tutorial-unpackaged-deployment?tabs=cpp) targets a console application and does not even get into how to deploy it.

I don't understand this documentation. What does WindowsAppSDKInstall.exe have to do with my unpackaged Win App SDK project? When I run WindowsAppSDKInstall.exe, all that is does is install some bogus app successfully (the assumption I get is that my dev environment is properly configured to deploy unpackaged apps).

[https://github.com/MicrosoftDocs/windows-uwp/blob/docs/hub/apps/windows-app-sdk/deploy-unpackaged-apps.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/hub/apps/windows-app-sdk/deploy-unpackaged-apps.md)

Run the Windows App SDK installer in your development environment
You can test deployment in your development environment by running the Windows App SDK Installer:

WindowsAppSDKInstall.exe if you are using version 1.0 Experimental or later
ProjectReunionInstall.exe if you are using version 0.8 Preview
You can also run the installer with no user interaction and suppress all text output with WindowsAppSDKInstall.exe --quiet or ProjectReunionInstall.exe --quiet.

**After the installation is complete, you can run your unpackaged app. For an example of how to build and run an unpackaged app that uses the Windows App SDK, see Tutorial: Build and deploy an unpackaged app that uses the Windows App SDK.**
---------------------- ^^^^^^^ What is this? ^^^^^^^ ----------------------

**Comment by andrewleader** [comment, marked as answer]
Hey @capegreg, thanks for asking this!

UWP apps cannot be unpackaged, UWP apps are always packaged.

Unpackaged only applies to desktop-based app types (like WPF and WinForms) - those that choose not to use MSIX.

The Windows App SDK operates similar to .NET, in that the computer has to have the Windows App Runtime installed for your app to use it. Your app does not carry the runtime assets itself. The runtime is shared across 1-to-many apps on the computer.

Therefore, for unpackaged apps, **when you're distributing your app to other computers**, you need to include the WindowsAppSDKInstall.exe, and run the installer along with the installation of your app. Packaged apps automatically install this dependency when their MSIX is installed.

**Reply by capegreg** [reply]
Thanks, Andrew. A couple of months ago there was no reason for needing an unpackaged app, but my company locked down MS Store which meant that I could no longer enable auto updatable packages in my UWP app. This made it tough to push updates to 70 PCs, and it got a lot of push back from the deployment admin when I said his techs would need to manually re-install the app to 70 PCs due to this new group policy. Had this information been communicated to me before development, I would have gone directly to WinUI or even Vue.js with Electron packaging. But I found out the hard way after testing an update release. The .appxbundle installer failed with Access is Denied. (On top of that, there is some weirdness where a perfectly working app later fails to launch after a Windows update. I'm still troubleshooting this.) Since WinUI 3 1.0.0-experimental1 release includes unpackaged app feature, I thought to migrate my UWP app to WinUI 3 but so far I have not been successful. My app is already in production, but enhancements and bug fixes are on hold until I figure out a new path. I am sort of forced to convince my company that MS Store needs to be open.

**Reply by andrewleader** [reply]
Thanks for the additional info!

While we have some unpackaged desktop app support in our Experimental release, WinUI 3 is not yet supported by unpackaged apps, but that is expected to ship in our 1.0 Stable release. So right now, you cannot build a WinUI 3 unpackaged desktop app. You can use WinAppSDK in an unpackaged WPF app and access the activation APIs, but that wouldn't be helpful for your scenario.

You should be able to distribute your UWP app outside of the store as you're trying to. You will have to use your own certificate to sign the app, and also get that certificate installed on the PCs you're trying to distribute to, before you can install those apps. Sideloading must be enabled on the computers too... it's typically enabled but some enterprises can disable that. Here's some more info on that: https://www.windowscentral.com/how-enable-windows-10-sideload-apps-outside-store

Have you tried those things?

**Reply by capegreg** [reply]
Yes. The app is sideloaded on all the machines, and developer mode is enabled. The cert is self-signed and imported into cert store. We are currently using elevated Powershell with assigned access user account to install. Do you think I can overcome this issue moving it to Azure? I’m not up to speed with Azure.

This is a SignalR app with .Net 5 \ MySQL Pomelo EF 6 backend. My requirement was Windows 10 desktop. I built the POC in Vue.js but later changed to UWP because I found packaging with Electron lacked any standards and the size, nearly 500mb too big. I really do not want to redevelop this in another technology if only I can get the auto update working or figure out how to push updates without doing manually. I am working with our security team to troubleshoot it, but being a high intel gov agency, security policies are not very flexible. I am in the design phase of a second app to be deployed to about 10 users, also Windows 10 desktop. I am considering WinUI 3 this time.

**Reply by andrewleader** [reply]
@jvintzel can you help @capegreg with how to enable automatic updates for a sideloaded enterprise UWP app?

**Reply by capegreg** [reply]
Update
I was able to get automatic updates working on a brand new Win10 machine that was not added to our domain. The install and update work great. Next we will add the machine to the domain and the install should break, which means we need to identify whatever group policy is preventing this from working.
